op-batcher/batcher/channel_manager.go

```diff
@@ -146,10 +150,47 @@ func (s *channelManager) nextTxData(channel *channel) (txData, error) {
 
 // TxData returns the next tx data that should be submitted to L1.
 //
-// If the pending channel is
+// If the current channel is
 // full, it only returns the remaining frames of this channel until it got
 // successfully fully sent to L1. It returns io.EOF if there's no pending tx data.
+//
+// It will generate the tx data internally, and decide whether to switch DA type
+// automatically. When switching DA type, the channelManager state will be rebuilt
+// with a new ChannelConfig.
 func (s *channelManager) TxData(l1Head eth.BlockID) (txData, error) {
+	data, err := s.txData(l1Head)
+	if s.currentChannel == nil {
+		s.log.Trace("no current channel")
+		return data, err
+	}
+	assumedBlobs := s.currentChannel.cfg.UseBlobs
```

Comment:
Couldn't we compare to the `s.defaultCfg` here?

Reply:
e299a7922d7d8ef54710b8c234386ce41fd26b69